src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/FunctionProbes/FunctionProbesManager.cs

```diff
@@ -94,14 +223,23 @@ public void StartCapturing(IList<MethodInfo> methods)
                 }
 
                 FunctionProbesStub.InstrumentedMethodCache = new ReadOnlyDictionary<ulong, InstrumentedMethod>(newMethodCache);
+
+                _installationTaskSource = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
This looks like it could race with a Dispose call. For example, the first set of state checks could succeed, dispose could be called, and then a new _installationTaskSource could be created.